### PR TITLE
Add snapshot script

### PR DESCRIPTION
**Pull request by sagikazarmark**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | 
| License         | Apache 2.0

### What's in this PR?
A new snapshot script for creating snapshots instead of the old git tag workflow.

### Why?
We don't want to spam the repo with tags created for internal testing.

### CODE REVIEW

File: BUILD.plz [hunk 1]
```diff
@@ -61,6 +61,171 @@ sh_cmd(
...
+        ;;
+        *)
+        ARGS+=("\\\$1")
+        shift # Remove generic argument from processing
+        ;;
+    esac
+done
+
+# Setting here so the last shift won't fail the script
+set -o errexit
```
Comment (pregnor) [inline BUILD.plz:143]: **Optional**: we **MAY** consider moving this closer to the `# Auto-detect tag (if possible)` conditional statement to make modifications to the `plz` command less error-prone thus avoiding the introduction of non-failing pipe commands between the setting and the mentioned conditional statement when those new commands are supposed to fail on pipe failure, but the author and reviewer miss the setting.

In addition I would consider turning pipefail off for the mentioned conditional block only and reenable pipefail right after that.

File: BUILD.plz [hunk 2]
```diff
@@ -61,6 +61,171 @@ sh_cmd(
...
+        fi
+
+        TAG="\\\$($(out_location ///pleasings2//tools/misc:semver) bump patch \\\$LATEST_VERSION)-snapshot.\\\\$(date +%s)"
+    elif [[ "\\\$CURRENT_BRANCH" == "master" ]]; then
+        LATEST_VERSION="\\\$(git tag | grep -E "\\\$VERSION_REGEX" | sort -r --version-sort | head -1)"
+        if [[ "\\\$LATEST_VERSION" == "" ]]; then
+            error "failed to determine latest version"
+        fi
+
+        TAG="\\\$($(out_location ///pleasings2//tools/misc:semver) bump minor \\\$LATEST_VERSION)-snapshot.\\\\$(date +%s)"
```
Comment (pregnor) [inline BUILD.plz:188]: **Note**: post 1.0.0 we could introduce a flag in this command for major version bumped snapshots and conditionally branch on it here.

**Review by pregnor — APPROVED** [review]

File: BUILD.plz (hunk 1 above)
Comment (orymate) [inline BUILD.plz:143]: The `shift`s should not fail in the current code (and setting `-e` earlier would detect errors like when you run the script with `--ref` but no argument to it). So I'd set it at the beginning.

File: BUILD.plz (hunk 2 above)
Comment (sagikazarmark) [reply BUILD.plz]: Done

File: BUILD.plz (hunk 1 above)
Comment (sagikazarmark) [reply BUILD.plz]: Done

**Review by pregnor — APPROVED** [review]

**Review by orymate — APPROVED** [review]

**Review by asdwsda — APPROVED** [review]

**Pull request merged**